packages/ai-ide/src/browser/template-preference-contribution.ts

```diff
@@ -0,0 +1,118 @@
+// *****************************************************************************
+// Copyright (C) 2024 EclipseSource GmbH.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import { FrontendApplicationContribution, PreferenceService } from '@theia/core/lib/browser';
+import { inject, injectable } from '@theia/core/shared/inversify';
+import { FrontendPromptCustomizationServiceImpl } from '@theia/ai-core/lib/browser/frontend-prompt-customization-service';
+import {
+    PROMPT_TEMPLATE_WORKSPACE_DIRECTORIES_PREF,
+    PROMPT_TEMPLATE_WORKSPACE_EXTENSIONS_PREF,
+    PROMPT_TEMPLATE_WORKSPACE_FILES_PREF
+} from './workspace-preferences';
+import { WorkspaceService } from '@theia/workspace/lib/browser';
+import { Path } from '@theia/core';
+
+@injectable()
+export class TemplatePreferenceContribution implements FrontendApplicationContribution {
```

Comment:
This contribution should also manage the other template customization directory preference which is currently part of the `ai-core-preferences`.

Reply:
Ah I didn't realize that this was only ide-specific. Is there a reason for this?

Reply:
I would not put it into a core package, as it enables the user to configure these directories. Adopters might not want this in other tools. So I claim the preferences are tool/IDE specific